src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -616,7 +619,7 @@ private void ProduceStart(bool appCompleted)
 
         protected Task TryProduceInvalidRequestResponse()
         {
-            if (_requestProcessingStatus == RequestProcessingStatus.RequestStarted && _requestRejected)
```

Comment:
I've added the test, but this wouldn't happen even without the change to `FrameOfT`. `SetBadRequestState` sets `_keepAlive` to `false`, which means the loop would exit here:

```
if (!_keepAlive)
{
    // End the connection for non keep alive as data incoming may have been thrown off
    return;
}
```

So `Reset()` would not get called in that situation.

The reason I had to add

```
// Don't lose request rejection state
if (!_requestRejected)
{
    Reset();
}
```

to `FrameOfT` is that the timeout happens before the app starts, and we never hit the check above.